crates/matrix-sdk-crypto/src/identities/user.rs

```diff
@@ -465,7 +596,15 @@ impl ReadOnlyUserIdentity {
     ) -> Result<bool, SignatureError> {
         master_key.verify_subkey(&self_signing_key)?;
 
-        let new = Self::new(master_key, self_signing_key)?;
+        // The pin is maintained.
```

Comment:
Perhaps we can expand this a bit, to explain the why. I know the why right now but another reader might not know it.